### PR TITLE
Bump @codemirror packages to version 6.x

### LINKED ISSUE #10458: Text is Reversed when typing in the Code Editor when Editing Lovelace Cards
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

On Windows 10, Firefox 93, when editing a Lovelace Card in the UI Code Editor, sometimes what I type is reversed, or entered right to left. Much of the time, it works as expected, but at other times, when I type 'livingroom', I get 'moorgnivil', for example. Even the `Del` and `Backspace` keys works opposite in those instances. Very frustrating and odd. Has happened on multiple occasions, and I can't determine a pattern, although, it does sometimes fix itself after going out and back in a few times. This NEVER happens in any other web page I use with Firefox, ONLY with HA.

### Describe the behavior you expected

That text is entered in the order it was entered, left to right, instead of right to left.

### Steps to reproduce the issue

Not sure how to reproduce this on demand. I don't know if it is Firefox or HA code. But it ONLY happens on the HA Frontend _when editing Cards in the Code Editor_. Nowhere else in Firefox, or in HA, does this occur, which points to the HA Web Frontend code somewhere.

### What version of Home Assistant Core has the issue?

2021.10.6

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Firefox 93

### Which operating system are you using to run this browser?

Win 10 x64

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by steverep**
## Proposed change

This is meant to replace #14783 from dependabot.  Nearly all of the packages use dependencies, instead of more accurately using peer dependencies, so these really should be updated as an entire scope when jumping major versions to avoid issues.  Apparently they are cautiously making the switch though (https://github.com/codemirror/dev/issues/858). 👍🏻 

Most of the changes needed were just to account for package consolidations and a few minor API changes.  The only thing I was unsure of is noted in the comment below.

This might close the issues mentioned below, or at least should be the first step to solving them.

## Type of change

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
  - #14134
  - #12982
  - #10458 
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-code-editor.ts [hunk 1]
```diff
@@ -152,10 +153,8 @@ export class HaCodeEditor extends ReactiveElement {
         saveKeyBinding,
       ] as KeyBinding[]),
       this._loadedCodeMirror.langCompartment.of(this._mode),
-      this._loadedCodeMirror.theme,
-      this._loadedCodeMirror.Prec.fallback(
```
Comment (steverep) [inline src/components/ha-code-editor.ts:156]: It's not clear to me why the precedence for the syntax highlighting was lowered.  If this was to account for a bug, there was some fixes for highlighting precedence in the release notes, but I don't know what to check for in HA.  I made the assumption it was not needed anymore for now.

**Review by bramkragten — APPROVED** [review]
Looks good! 👍

**Comment by WRoss7** [comment]
Persistent in 2023.2.0

**Comment by steverep** [comment]
@WRoss7 please add your comment to whichever issue you are referring.  Thanks!

**Pull request merged**